STORM-1097. Compile logical plans to Java source code.

haohui opened the pull request:
This PR implements the compilers that compile logical plans from Calcite to Java source code. It also removes the compiler implemented in native code per discussions in https://issues.apache.org/jira/browse/STORM-1105?focusedCommentId=14970164&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14970164.

harshach commented on external/sql/storm-sql-core/src/jvm/org/apache/storm/sql/compiler/ExprCompiler.java:
```diff
@@ -0,0 +1,210 @@
+        return printer.translate(compiler, call);
+      }
+    }
+    private Map.Entry<SqlOperator, CallExprPrinter> infixBinary
```
new line here.

harshach commented on external/sql/storm-sql-core/src/jvm/org/apache/storm/sql/compiler/ExprCompiler.java:
```diff
@@ -0,0 +1,210 @@
+    String translate(ExprCompiler compiler, RexCall call);
+  }
+
+  private static class ImpTable {
```
what is ImpTable can you add some details here.

harshach commented:
@haohui overall looks good to me. couple of minor nitpicks.

haohui commented:
Thanks @harshach for the reviews.

ImpTable maps the SQL operators to the classes that understand the SQL operators and generate corresponding Java source code.

I updated the patch and added comments for the class.

harshach commented:
+1
